test/compeon/rack_tools/pipes/extract_parameters_from_request_test.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+require 'compeon/rack_tools/pipes/extract_parameters_from_request'
+
+module Compeon
+  module RackTools
+    module Pipes
+      class ExtractParametersTest < Minitest::Test
+        def test_with_present_params
+          request = Rack::Request.new(
+            Rack::RACK_INPUT => '',
+            Rack::QUERY_STRING => 'param=present&param2=another-param'
+          )
+          extractor = Compeon::RackTools::Pipes::EXTRACT_PARAMETERS_FROM_REQUEST.call(parameter_names: %w[param param2])
+
+          assert_equal(extractor.call(request), request: request, param: 'present', param2: 'another-param')
+        end
+
+        def test_with_missing_params
+          request = Rack::Request.new(
+            Rack::RACK_INPUT => '',
+            Rack::QUERY_STRING => 'param=present'
+          )
+          extractor = Compeon::RackTools::Pipes::EXTRACT_PARAMETERS_FROM_REQUEST.call(parameter_names: %w[param param2])
+
+          assert_raises Compeon::RackTools::UnprocessableEntityError do
+            extractor.call(request)
+          end
+        end
+
+        def test_with_unrequired_params
+          request = Rack::Request.new(Rack::RACK_INPUT => '', Rack::QUERY_STRING => 'param=present')
```

Comment:
```suggestion
          request = Rack::MockRequest.env_for('', params: { param: 'present' })
```

Reply:
This is an env that you also use to test `PARSE_ENV`. Why would I use that and not work with the request object that `PARSE_ENV` spits out?